SwiftReflector/SwiftXmlReflection/TypeDeclaration.cs

```diff
@@ -222,6 +222,17 @@ public static TypeDeclaration TypeFromXElement (XElement elem, ModuleDeclaration
 					var assocElements = from assocElem in elem.Element ("associatedtypes").Elements ()
 							    select AssociatedTypeDeclaration.FromXElement (assocElem);
 					protoDecl.AssociatedTypes.AddRange (assocElements);
+					var selfIndex = protoDecl.AssociatedTypes.FindIndex (at => at.Name == "Self");
```

Comment:
Are there other safeguards to ensure that there's never more than one "Self" in protodecl?

Reply:
Good one!

Reply:
I don't think there _will_ be that case, but using `RemoveAll` on the list neatens up the code quite a bit.